teuthology/suite/test/test_run_.py

```diff
@@ -111,3 +114,195 @@ def test_sha1_nonexistent(
+        m_get_package_versions,
+        m_has_packages_for_distro,
+        m_schedule_jobs,
+        m_find_git_parent,
+    ):
+        m_git_validate_sha1.return_value = self.args.ceph_sha1
+        m_package_version_for_hash.return_value = 'ceph_version'
+        m_git_ls_remote.return_value = 'suite_hash'
+        build_matrix_desc = 'desc'
+        build_matrix_frags = ['frag.yml']
+        build_matrix_output = [
+            (build_matrix_desc, build_matrix_frags),
+        ]
+        m_build_matrix.return_value = build_matrix_output
+        m_file.side_effect = [StringIO('field: val\n') for i in xrange(11)]
+        m_get_install_task_flavor.return_value = 'basic'
+        m_get_package_versions.return_value = dict()
+        m_has_packages_for_distro.side_effect = [
+            False for i in xrange(11)
+        ]
+
+        m_find_git_parent.side_effect = lambda proj, sha1: sha1 + '^'
+
+        self.args.newest = 10
+        runobj = self.klass(self.args)
+        runobj.base_args = list()
+        with pytest.raises(ScheduleFailError) as exc:
+            runobj.schedule_suite()
+        assert 'Exceeded 10 backtracks' in str(exc.value)
+        m_find_git_parent.assert_has_calls(
+            [call('ceph', 'ceph_sha1' + i * '^') for i in xrange(10)]
+        )
+
+    @patch('teuthology.suite.util.find_git_parent')
+    @patch('teuthology.suite.run.Run.schedule_jobs')
+    @patch('teuthology.suite.util.has_packages_for_distro')
+    @patch('teuthology.suite.util.get_package_versions')
+    @patch('teuthology.suite.util.get_install_task_flavor')
+    @patch('__builtin__.file')
+    @patch('teuthology.suite.run.build_matrix')
+    @patch('teuthology.suite.util.git_ls_remote')
+    @patch('teuthology.suite.util.package_version_for_hash')
+    @patch('teuthology.suite.util.git_validate_sha1')
+    def test_newest_success(
+        self,
+        m_git_validate_sha1,
+        m_package_version_for_hash,
+        m_git_ls_remote,
+        m_build_matrix,
+        m_file,
+        m_get_install_task_flavor,
+        m_get_package_versions,
+        m_has_packages_for_distro,
+        m_schedule_jobs,
+        m_find_git_parent,
+    ):
+        m_git_validate_sha1.return_value = self.args.ceph_sha1
+        m_package_version_for_hash.return_value = 'ceph_version'
+        m_git_ls_remote.return_value = 'suite_hash'
+        build_matrix_desc = 'desc'
+        build_matrix_frags = ['frag.yml']
+        build_matrix_output = [
+            (build_matrix_desc, build_matrix_frags),
+        ]
+        m_build_matrix.return_value = build_matrix_output
+        m_file.side_effect = [StringIO('field: val\n') for i in xrange(11)]
+        m_get_install_task_flavor.return_value = 'basic'
+        m_get_package_versions.return_value = dict()
+        m_has_packages_for_distro.side_effect = \
+            [False for i in xrange(5)] + [True]
+
+        m_find_git_parent.side_effect = lambda proj, sha1: sha1 + '^'
+
+        self.args.newest = 10
```

Comment:
it's less than 10.  It tries for some number of parent commits and then succeeds.